src/core/MeshGenerator.ts

```diff
@@ -42,29 +41,47 @@ export class MeshGenerator {
     this.entity = entity;
   }
 
-  initialize(skeleton?: Skeleton) {
+  initialize(skeletonData: SkeletonData, setting?: SpineRenderSetting) {
+    if (!skeletonData) return;
+
     const meshRenderer = this.entity.getComponent(MeshRenderer);
     if (!meshRenderer) {
       console.warn('You need add MeshRenderer component to entity first');
       return;
     }
-    const vertexCount = this.getVertexCount(skeleton);
-    this.spineMesh.initialize(this.engine, vertexCount);
-    meshRenderer.mesh = this.spineMesh.mesh;
-  }
-
-  buildMesh(skeleton: Skeleton, setting?: SpineRenderSetting) {
-    if (!skeleton) {
-      return;
-    }
 
     if (setting) {
       this.setting = setting;
     }
 
-    const meshRenderer = this.entity.getComponent(MeshRenderer);
-    if (!meshRenderer) {
-      console.warn('You need add MeshRenderer component to entity first');
+    // Prepare buffer by using all attachment data but clippingAttachment
+    const { defaultSkin: { attachments } } = skeletonData;
+    let vertexCount: number = 0;
+    const attachmentsArray = [];
+    for (let i = 0, n = attachments.length; i < n; i++) {
+      for (let key in attachments[i]) {
+        attachmentsArray.push(attachments[i][key]);
+      }
+    }
+    for (let i = 0, n = attachmentsArray.length; i < n; i++) {
+      const attachment = attachmentsArray[i];
+      if (!attachment) {
+        continue;
+      } else if (attachment instanceof RegionAttachment) {
+        vertexCount += MeshGenerator.QUAD_TRIANGLES.length;
```

Comment:
MeshGenerator.QUAD_TRIANGLES.length cache before for is better? the same as mesh.triangles.length